python/lsst/pex/config/config.py

```diff
@@ -60,6 +67,18 @@
         YamlLoaders += (CLoader,)
     except ImportError:
         pass
+else:
+    YamlLoaders = ()
+    doImport = None
```

Comment:
Isn't this already set from the try/except earlier?  I'm also a bit surprised it's not assigned to something else when `yaml` is available, but maybe that will be explained by code below.

Reply:
I feel that this change should be a separate commit. It's probably better to move the setting down from the import check above but it doesn't seem directly related to the other changes.

Reply:
We can move it or drop it. This just fell under "a few other changes to make type checking happy". My checker was unable to tell that `YamlLoaders` would be bound, and was complaining about a possibly unbound variable, which is understandable as it would need to know about the logic in the exception handler triggering setting variables that will later be used in an `if` statement check.